src/components/dataDisplays/cellRenderers/OverflowController.jsx

```diff
@@ -0,0 +1,61 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { debounce } from 'lodash-es';
+
+import Tooltip from '@material-ui/core/Tooltip';
+
+const noWrapStyles = {
+  display: 'block',
+  whiteSpace: 'nowrap',
+  overflow: 'hidden',
+  textOverflow: 'ellipsis',
+};
+
+export default function OverflowController({ title, children }) {
+  const [isOverflowed, setIsOverflowed] = useState(false);
+  const [refEl, setRefEl] = useState();
+
+  const ref = useCallback(node => {
+    setRefEl(node);
+  }, []);
+
+  useEffect(
+    () => {
+      if (!refEl) return;
+
+      const debouncedHandleResize = debounce(function handleResize() {
+        const { clientWidth, scrollWidth } = refEl;
+        const overflow = scrollWidth - clientWidth > 0;
+        setIsOverflowed(overflow);
+      }, 200);
+
+      const resizeObserver = new ResizeObserver(
+        debouncedHandleResize,
+      );
+      resizeObserver.observe(refEl);
+
+      return function unobserveResizeObserver() {
+        resizeObserver.unobserve(refEl);
+      };
+    },
+    [refEl],
+  );
+
+  const childrenWithStyles = React.cloneElement(
+    React.Children.only(children),
+    {
+      ref,
+      style: noWrapStyles,
```

Comment:
would this override styles being passed to the children? if so we might need to merge noWrapStyles with any other styles

Reply:
Yes! Also, if a ref is passed to the child, this new ref is ignored. Currently none of the cell renderers accept style props or refs though.